airbyte-server/src/main/java/io/airbyte/server/ServerApp.java

```diff
@@ -233,6 +233,10 @@ public static ServerRunnable getServer(final ServerFactory apiFactory,
     final Flyway jobsFlyway = FlywayFactory.create(jobsDataSource, DbMigrationHandler.class.getSimpleName(), JobsDatabaseMigrator.DB_IDENTIFIER,
         JobsDatabaseMigrator.MIGRATION_FILE_LOCATION);
 
+    // It is important that the migration to the temporal scheduler is performed before the server accepts any requests.
+    // This is why this migration is performed here instead of in the bootloader - so that the server blocks on this.
+    migrateExistingConnectionsToTemporalScheduler(configRepository, jobPersistence, eventRunner);
```

Comment:
Recall that as I described in [this comment](https://github.com/airbytehq/airbyte/pull/12589#discussion_r865464860), the logic to cleanup non-terminal jobs is only executed just before a new job is created by the temporal workflow.

One consequence of this is that if a user upgrades while a job is running, then that job will continue to have that `RUNNING` status until the _next_ time that the connection is synced. So in the worst case, this means 24 hours could pass before that zombie job is marked as `FAILED` and a new job is created.

This isn't an ideal situation, but I think it is tolerable as a one-time occurrence for this migration.

Reply:
Just to note, this was fixed in that other PR by also cleaning up jobs at the beginning of the temporal workflow, so this should no longer be an issue